goalc/compiler/IR.cpp

```diff
@@ -1274,27 +1295,42 @@ void IR_RegSetAsm::do_codegen(emitter::ObjectGenerator* gen,
 // AsmVF3
 ///////////////////////
 
+// TODO - make a varient that doesn't take a mask
+
 IR_VFMath3Asm::IR_VFMath3Asm(bool use_color,
                              const RegVal* dst,
                              const RegVal* src1,
                              const RegVal* src2,
+                             u8 mask,
                              Kind kind)
-    : IR_Asm(use_color), m_dst(dst), m_src1(src1), m_src2(src2), m_kind(kind) {}
+    : IR_Asm(use_color), m_dst(dst), m_src1(src1), m_src2(src2), m_mask(mask), m_kind(kind) {}
 
 std::string IR_VFMath3Asm::print() {
+  std::string function = "";
   switch (m_kind) {
     case Kind::XOR:
-      return fmt::format(".xor.vf{} {}, {}, {}", get_color_suffix_string(), m_dst->print(),
-                         m_src1->print(), m_src2->print());
+      function = ".xor.vf";
+      break;
     case Kind::SUB:
-      return fmt::format(".sub.vf{} {}, {}, {}", get_color_suffix_string(), m_dst->print(),
-                         m_src1->print(), m_src2->print());
+      function = ".sub.vf";
+      break;
     case Kind::ADD:
-      return fmt::format(".add.vf{} {}, {}, {}", get_color_suffix_string(), m_dst->print(),
-                         m_src1->print(), m_src2->print());
+      function = ".add.vf";
+      break;
+    case Kind::MUL:
+      function = ".mul.vf";
+      break;
+    case Kind::MAX:
+      function = ".max.vf";
+      break;
+    case Kind::MIN:
+      function = ".min.vf";
+      break;
     default:
       assert(false);
   }
+  return fmt::format("{}{} {}, {}, {}, {}", function, get_color_suffix_string(), m_dst->print(),
```

Comment:
For printing the mask, can we do `{:#b}`? It will print it as binary with the `0b` prefix, which will be easier to read than a decimal.